src/Tickets/Commerce/Flag_Actions/Archive_Attendees.php

```diff
@@ -40,8 +40,9 @@ class Archive_Attendees extends Flag_Action_Abstract {
 	 * {@inheritDoc}
 	 */
 	public function handle( Status_Interface $new_status, $old_status, \WP_Post $post ) {
-		// @todo we need an error handling piece here.
-		if ( empty( $post->items ) ) {
+		$status_args = $new_status->get_wp_arguments();
+
+		if ( empty( $post->items ) || $status_args['label'] !== $post->status_name ) {
```

Comment:
I dont follow this check.

Reply:
This is in place to make sure that the status we're moving to in `$new_status` is the same as we already have in the order. So we don't make any changes to attendees if those two mismatch for some reason.